### PR TITLE
feat: eth command

### LINKED ISSUE #72: send coins and query balance of address
Many times occurred when generated addresses required fund to perform some transaction. It was solved by

### PR DESCRIPTION
**Pull request by nugaon**
Added `eth` group command with
- `balance`
- `send`
commands.

The test environment got updated with command log related utility function, namely `describeCommand` and other changes regarding how the error handling happens in the CLI.

Resolves #72

### CODE REVIEW

File: src/command/eth/eth-command.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { RootCommand } from '../root-command'
+import { providers } from 'ethers'
+import { CommandLineError } from '../../utils/error'
+
+export class EthCommand extends RootCommand {
+  protected provider = new providers.JsonRpcProvider('http://localhost:9545')
```
Comment (IgorShadurin) [inline src/command/eth/eth-command.ts:6]: Does it need to be hardcoded? Maybe it's better to put it in env?

File: src/command/eth/send.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    type: 'number',
+    description: 'The account index from which the coins will be sent.',
+    default: 0,
+  })
+  public from!: number
+
+  public async run(): Promise<void> {
+    await super.init()
+
+    // this.assertAddress(this.to)
```
Comment (IgorShadurin) [inline src/command/eth/send.ts:40]: Commented code could be removed

File: test/integration/eth.spec.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-console */
+import crypto from 'crypto'
+import * as ethers from 'ethers'
+
+import { run } from '../utils/run'
+import { ENV_ENV_PREFIX_KEY } from '../../src/constants'
+import { describeCommand } from '../utils/console-log'
+import { sleep } from '../../src/utils'
+
+describeCommand('stop command', ({ getLastMessage, getNthLastMessage }) => {
```
Comment (IgorShadurin) [inline test/integration/eth.spec.ts:10]: 'getNthLastMessage' is defined but never used

File: src/command/eth/eth-command.ts (hunk 1 above)
Comment (nugaon) [reply src/command/eth/eth-command.ts:6]: you cannot change URL parameters in the CLI so it is always contant on the host.
Nevertheless I defined it in the constant.ts instead

File: src/command/eth/eth-command.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+  protected async init(): Promise<void> {
+    await super.init()
+
+    const network = await this.provider.getNetwork()
+
+    if (network.chainId !== 4020) {
```
[the remark this answers is not in the document]
Comment (nugaon) [reply src/command/eth/eth-command.ts:13]: it is also fixed, but I will put into the constant.ts

**Review by IgorShadurin — APPROVED** [review]
LGTM!

**Comment by nugaon** [comment]
Release [v2.2.0](https://github.com/fairDataSociety/fdp-play/releases/tag/v2.2.0) addresses this.

**Pull request merged**